app/controllers/idv/image_uploads_controller.rb

```diff
@@ -48,7 +48,12 @@ def render_form_response(form_response)
           success: true,
         }
       else
-        render json: form_response.to_h,
+        # FOLLOWUP: refactor the client to accept a hash of errors
+        errors_arr = form_response.errors.flat_map do |key, errs|
+          Array(errs).map { |err| "#{key} #{err}" }
+        end
+
+        render json: form_response.to_h.merge(errors: errors_arr),
                status: :bad_request
```

Comment:
cc @aduth the underlying change here means we now can have easier per-attribute errors for the image uploading API, so we could do something like:

```
errors: { selfie: ["some error"], front: ["some other error"] }
```

or one of these if we want:

```
errors: [
  { attribute: "front", message: "some error" },
]
```

So let me know if there's one you prefer and let's pair on refactoring the server + JS in the same PR?

Reply:
> So let me know if there's one you prefer and let's pair on refactoring the server + JS in the same PR?

Just to clarify, will those changes need to be made here? Trying to understand if we'll need to update [this logic](https://github.com/18F/identity-idp/blob/c4412a69e4eb0ca7b7fb4dce15bfcc088a19c82e/app/javascript/packages/document-capture/services/upload.js#L45-L46) if `errors` is no longer an array.

To the question, I don't have a strong preference either way. I feel like it's likelier we'll want to show the errors at the specific input where the value is assigned, so pre-grouping them by field name might make that a bit simpler, whereas the array form would be more useful if we expected to show all errors together.

Reply:
This PR preserves the existing API because (array of string errors), but the underlying changes let us create the per-field errors we had talked about before. I'm saying for a future PR, let's figure out what we want in the client and pair on building that